### PR TITLE
non-con headpats begone

### PR DESCRIPTION
**Pull request by goobliner**
## About The Pull Request

**COMPLETELY OVERHAULED (overstatement)**

Adds a button in the IC verbs section which toggles you being able to be hugged, nose booped or headpatted. Works on robots and humans and aliens and all of the above and what not.

works perfect

## How This Contributes To The Skyrat Roleplay Experience

Many people, myself included, don't like getting headpatted and hugged all the time.

## Proof of Testing

<details>
<summary></summary>

image

</details>

## Changelog

:cl:
add: Adds a new verb in the IC tab, "toggle reflexes", which lets your toggle being able to be head pat, nose bopped or hugged. Works for both people AND robots.
/:cl:

### CODE REVIEW

**Comment by W4rd3nn** [comment]
update ur stupid byond dinglebrain

**Comment by goobliner** [comment]
> update ur stupid byond dinglebrain

but i really dont want to do i really have to

**Comment by Namwty** [comment]
I would highly request that the text isn't red or even better, not have visible text at all.
**Frankly it should just give "notice" type msg to person who is trying to hug and him alone saying "You feel like you shouldn't hug the person" or whatever.**

1. It's annoying as hell to have red text fill the chat, oversized person here ( who already has this function due to trait, everytime someone tries to pat me, I basically dart to chat thinking something bad is going on ), this should be reserved to hostile actions that are generally something worse that should be "OH SHIT, LOOK SOMEONE MIGHT BE GETTING KILLED"

2. Seems like a potential drama bait for OOC shit of "Wow creep tried to hug me, get lost.", I don't care who is in right or wrong, it's ass for staff to deal with or any outsiders to see.

3. In fighting with fists, you sometimes happen to just forget to toggle combat mode, which means you will be hugging instead of punching. Now issue is, if people see red text when they are hugging person they are trying to punch... they will just assume that they are punching them ( cuz red text instead of blue ) or even worse, god forbid if someone reads "swatted away" crap as Sleeping Carp(CQC) ( as in swaying away bullets, in combat sometimes people get overwhelmed and don't have time to read chat ) meaning that someone might just be executed on spot.


Other shit include like checking body temp of someone without health scanner ( hugs can tell if someone is overheating or cold ), nightmare panic increase/decrease hugging interaction, but that all can be put as "quirk feature" and frankly this PR is good and brings more positive experience for players, it's just the red warning text shouldn't be a thing.

Sadly I doubt borgs can't have this ( as they are basically main people who get "OwO CUTE BORG *pats *licks *hugs" ), honestly maybe should have some type of option for borgs that allows this to function ( even if not in quirk format )

**Comment by goobliner** [comment]
I can make the message a private one, sure, makes sense.

Borgs can't have traits, sadly.

If the person is unconscious they can't swat your hands away and you'll be able to check if they're too hot or cold, btw.

**Comment by HoodMilk** [comment]
stealth nightmare nerf

**Comment by goobliner** [comment]
i dunno why runs are failing, I'll have to look into that

**Comment by W4rd3nn** [comment]
> i dunno why runs are failing, I'll have to look into that

dis is why 
image

**Comment by goobliner** [comment]
I didn't know you weren't allowed to use the same symbol. Guess that explains it, I'll draw something new when I have time tomorrow, thanks for pointing that out I've never been able to read linters right for some reason.

**Comment by GlaciaImca** [comment]
Just a question but would it be possible to make it a toggle some how rather then a quirk? 

Not only because of the aforementioned borg issue, but also what about in the cases where you actually _are_ open to a hug or a head-pat....

Like I know for some people its just going to be a no ever, but there are also cases where like.... your character might get really annoyed with random hugs and pats from strangers, but be more tolerable from a friend or a lover.... a way to lower guard around only trusted people would be nice... a quirk locks you into one stance the entire round as far as I know.

**Comment by burgerenergy** [comment]
We already have doesn't like to be touched or whatever, I feel like this is an unnecessary escalation to that and turning an incredibly ignorable IC thing into an OOC issue. I mean, like, sure I guess? But why?

**Comment by goobliner** [comment]
> Just a question but would it be possible to make it a toggle some how rather then a quirk?

I can look into this, put no promises.Anything is possible, I'm just not sure I have the know how. I have a general idea, but i won't know until I actually get back to my computer and look at it in a few hours. I will try though, I like this idea.

> why?

Some people don't like receiving headpats and hugs from strangers in game but they aren't really doing anything wrong so it's not really actionable. This is like a middle ground so to speak? Maybe there's a better word I dunno.

**Comment by Cyprex** [comment]
> I didn't know you weren't allowed to use the same symbol. Guess that explains it, I'll draw something new when I have time tomorrow, thanks for pointing that out I've never been able to read linters right for some reason.

Honestly I'm not sure why you're not allowed to use the same symbol either!
> Why

Because some people want it and don't perceive it as ignorable as we do!

Edit: I was linked this, thought this was a bubberstation PR, not a skyrat one- I'm obviously not a skyrat maintainer!

**Comment by Floofies** [comment]
@goobliner 
Does this also stop the pat, hug, and headpat emotes from the Interaction Panel / Emote Panel? I think they should also be added, since removing the click-based interactions does nothing to stop roleplay, and this is a HRP server.

Alternatively, keeping those emotes solves the issue posited by @GlaciaImca

**Comment by chemistrymain2** [comment]
> I can make the message a private one, sure, makes sense.
> 
> Borgs can't have traits, sadly.
> 
> If the person is unconscious they can't swat your hands away and you'll be able to check if they're too hot or cold, btw.

could you give cyborgs a toggleable verb that makes them unable to be hugged if it can't be a trait

**Comment by goobliner** [comment]
> could you give cyborgs a toggleable verb that makes them unable to be hugged if it can't be a trait

I'm actually looking into scrapping the trait idea as a whole and just doing this for both people and robots.

**Comment by goobliner** [comment]
> @goobliner Does this also stop the pat, hug, and headpat emotes from the Interaction Panel / Emote Panel? I think they should also be added, since removing the click-based interactions does nothing to stop roleplay, and this is a HRP server.
> 
> Alternatively, keeping those emotes solves the issue posited by @GlaciaImca

It doesn't but I'll look into making it so it does whenever I turn this into a toggleable preference. No promises as I fear the verb menu and how shit it's code is, even in dreammaker standards.

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (SomeRandomOwl) [inline code/modules/mob/living/carbon/carbon_defense.dm]: Make sure to properly encompass the sr edit 

```suggestion
		else // START OF SKYRAT ADDITION
			if (HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS) && (!src.handcuffed))
				visible_message(span_warning("[helper] tries to hug [src], hands are swatted away!"))
				return
			else
				helper.visible_message(span_notice("[helper] hugs [src] to make [p_them()] feel better!"), \
							null, span_hear("You hear the rustling of clothes."), DEFAULT_MESSAGE_RANGE, list(helper, src))
				to_chat(helper, span_notice("You hug [src] to make [p_them()] feel better!"))
				to_chat(src, span_notice("[helper] hugs you to make you feel better!"))
			        // END OF SKYRAT ADDITION
```

**Review by SomeRandomOwl — CHANGES_REQUESTED** [review]

**Comment by nikothedude** [comment]
> We already have doesn't like to be touched or whatever, I feel like this is an unnecessary escalation to that and turning an incredibly ignorable IC thing into an OOC issue. I mean, like, sure I guess? But why?

because theres no way to really stop people from doing it pre-emptively if you just really dont like it

same reason we have personal space quirk. we have precedent

**Comment by goobliner** [comment]
11 linters failed

coraline-daddy-chill

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (nikothedude) [inline code/modules/mob/living/carbon/carbon_defense.dm]: this will fail and send to nobody

also switch this back to the visible message, this is a v isible action

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (nikothedude) [inline code/modules/mob/living/carbon/carbon_defense.dm]: ditto, also, trie is a typo

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (goobliner) [reply code/modules/mob/living/carbon/carbon_defense.dm]: fguck yu duoeude

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (goobliner) [reply code/modules/mob/living/carbon/carbon_defense.dm]: noge noe no

**Comment by goobliner** [comment]
okay


everything works now

besides that one thing that doesnt work but i dont have a clue how to fix that

the issue is that whenever you join as a human it automatically fires the verb, therefore it turns on quick reflexes, which turns off headpats and hugs. this isnt that big of a deal and could honestly just be merged... because the toggle works normally, i just have no clue why it fires automatically.

everything else works though. speedmerge

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (SomeRandomOwl) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
                        //SKYRAT EDIT BEGIN - DISALLOWED HEADPATS
			if(HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS))
				visible_message(span_warning("[user] tries to pet [src], but it moves out of the way."))
				return
			else
				visible_message(span_notice("[user] pets [src]."), \
								span_notice("[user] pets you."), null, null, user)
				to_chat(user, span_notice("You pet [src]."))
				user.add_mood_event("pet_borg", /datum/mood_event/pet_borg)
			//SKYRAT EDIT END - DISALLOWED HEADPATS
```

Non modular edits need to be marked

**Review by SomeRandomOwl — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm
Comment (SomeRandomOwl) [inline modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm]:
```suggestion
/mob/living/verb/reflexes_verb()
    set name = "Toggle Reflexes"
    set category = "IC"
    if(!HAS_TRAIT_FROM(src, TRAIT_QUICKREFLEXES, REF(src)))
        ADD_TRAIT(src, TRAIT_QUICKREFLEXES, REF(src))
        to_chat(src, span_notice("[get_reflexes_gain_text()]"))
    else if(HAS_TRAIT_FROM(src, TRAIT_QUICKREFLEXES, REF(src)))
        REMOVE_TRAIT(src, TRAIT_QUICKREFLEXES, REF(src))
        to_chat(src, span_notice("[get_reflexes_lose_text()]"))

/mob/living/proc/get_reflexes_gain_text()
	return "You don't feel like being touched right now."

/mob/living/proc/get_reflexes_lose_text()
	return "You'll allow yourself to be touched now."

/mob/living/silicon/get_reflexes_gain_text()
	return "Our systems will disallow platonic contact."

/mob/living/silicon/get_reflexes_lose_text()
	return "Our systems will allow platonic contact."
```
Unnecessary as this is a modular file

**Review by SomeRandomOwl — DISMISSED** [review]

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (goobliner) [reply code/modules/mob/living/silicon/silicon_defense.dm]: how'd I let this one slip by me I could've swore I got all the marks down. I'm way too tired to be coding lol, thanks

**Comment by SomeRandomOwl** [comment]
oh ffs github edit suggest is making CI pissy for you, one sec

**Comment by SomeRandomOwl** [comment]
There you go, sorry about that

**Comment by goobliner** [comment]
> oh ffs github edit suggest is making CI pissy for you, one sec

I'll just roll back your changes real quick and add them myselves it'll only take a moment

**Comment by goobliner** [comment]
should be ready to merge now unless any maint has an issue. bugs are fixed and code's good.

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (tf-4) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			visible_message(span_warning("[helper] tries to pat [src] on the head, but [p_they()] move[p_s()] out of the way."))
```
Idk the actual helper I just made one up to illustrate my point

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (tf-4) [inline code/modules/mob/living/carbon/carbon_defense.dm]: Same here

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [reply code/modules/mob/living/carbon/carbon_defense.dm]: That's correct

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
				visible_message(span_warning("[helper] tries to hug [src], but [p_they()] move[p_s()] out of the way."))
```

File: modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm
Comment (vinylspiders) [inline modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm:18]:
```suggestion
    set category = "IC"
   
    if(!HAS_TRAIT(src, TRAIT_QUICKREFLEXES))
```
This can't be gained from any other source, right?

File: modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm
Comment (vinylspiders) [inline modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm]:
```suggestion
    else
```
No need to put a condition here, since it's implied. (think it looks a bit neater)

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
		//SKYRAT EDIT ADDITION BEGIN - OVERSIZED & DISALLOWED HEADPATS
```

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			// SKYRAT EDIT ADDITION START
```

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			if (HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS) && src.incapacitated(IGNORE_RESTRAINTS))
```
Wouldn't you still be able to move out of the way if you're handcuffed? Also `incapacitated()` would be better in this case.

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
			if(HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS) && src.incapacitated(IGNORE_RESTRAINTS))
```
Did you want this to be the same as the other?

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
		else // SKYRAT EDIT ADDITION START
```

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
			else // SKYRAT EDIT ADDITION END
```

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			// SKYRAT EDIT ADDITION END
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (goobliner) [reply code/modules/mob/living/carbon/carbon_defense.dm]: Probably, I forgot to change that out earlier, I'll do that now thanks for reminding me.

File: modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm
Comment (goobliner) [reply modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm]: true

File: modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm
Comment (goobliner) [reply modular_skyrat/modules/modular_items/lewd_items/code/verbs.dm:18]: I was told to future proof it, no harm in doing so.

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
		else if(HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS) && !src.incapacitated(IGNORE_RESTRAINTS))
```

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			if (HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS) && !src.incapacitated(IGNORE_RESTRAINTS))
```

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
			if(HAS_TRAIT(src, TRAIT_QUICKREFLEXES) && (src.stat != UNCONSCIOUS) && !src.incapacitated(IGNORE_RESTRAINTS))
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]

**Review by vinylspiders — APPROVED** [review]

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
		else // SKYRAT EDIT ADDITION START
```

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			// SKYRAT EDIT ADDITION START
```

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (goobliner) [reply code/modules/mob/living/silicon/silicon_defense.dm]: damn I'm really letting these things slip by me, YIKES!!!!

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (goobliner) [reply code/modules/mob/living/carbon/carbon_defense.dm]: YIKES!!!!

File: code/modules/mob/living/carbon/carbon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
			// SKYRAT EDIT ADDITION END
			helper.visible_message(span_notice("[helper] hugs [src] to make [p_them()] feel better!"), \
						null, span_hear("You hear the rustling of clothes."), DEFAULT_MESSAGE_RANGE, list(helper, src))
			to_chat(helper, span_notice("You hug [src] to make [p_them()] feel better!"))
			to_chat(src, span_notice("[helper] hugs you to make you feel better!"))
```
Last thing

File: code/modules/mob/living/silicon/silicon_defense.dm
Comment (vinylspiders) [inline code/modules/mob/living/silicon/silicon_defense.dm]:
```suggestion
			// SKYRAT EDIT ADDITION END
			visible_message(span_notice("[user] pets [src]."), \
							span_notice("[user] pets you."), null, null, user)
			to_chat(user, span_notice("You pet [src]."))
			user.add_mood_event("pet_borg", /datum/mood_event/pet_borg)
```
This too

**Comment by vinylspiders** [comment]
I fixed it for you this time but in future just keep in mind when you indent blocks of tg code that is still considered a Skyrat Edit.

**Comment by goobliner** [comment]
> I fixed it for you this time but in future just keep in mind when you indent blocks of tg code that is still considered a Skyrat Edit.

Got it, thank you

**Review by vinylspiders — APPROVED** [review]

**Pull request merged**
